packages/gallery/src/components/gallery/proGallery/slideshowView.js

```diff
@@ -208,193 +226,187 @@ class SlideshowView extends GalleryComponent {
         avoidIndividualNavigation,
         ignoreScrollPosition,
         isContinuousScrolling,
+        scrollingUpTheGallery,
       });
     }
     this.removeArrowsIfNeeded();
   }
 
+  getNextItemOrGroupToScrollTo(
+    initiator,
+    direction,
+    ignoreScrollPosition,
+    avoidIndividualNavigation,
+    isAutoTrigger
+  ) {
+    let nextIndex;
+    if (initiator === 'nextItem') {
+      if (ignoreScrollPosition) {
+        nextIndex = this.state.activeIndex;
+      } else {
+        if (
+          avoidIndividualNavigation &&
+          !(this.props.styleParams.groupSize > 1)
+        ) {
+          nextIndex = this.getCenteredItemIdxByScroll();
+        } else {
+          nextIndex = isAutoTrigger
+            ? this.setCurrentItemByScroll()
+            : this.state.activeIndex;
+        }
+      }
+      nextIndex += direction;
+
+      if (!this.props.styleParams.slideshowLoop) {
+        nextIndex = Math.min(
+          this.props.galleryStructure.items.length - 1,
+          nextIndex
+        );
+        nextIndex = Math.max(0, nextIndex);
+      }
+    } else if (initiator === 'nextGroup') {
+      nextIndex = this.getCenteredGroupIdxByScroll() + direction;
```

Comment:
These functions are duplicated 
`getCenteredItemIdxByScroll`
`getCenteredGroupIdxByScroll`

They are doing the same thing with different variable names. Please use one of them and delete the other function

Reply:
@nirnaor It's not exactly the same thing, but I deleted one of them and combined the different logics

Reply:
ok @yakovhaiilo let me know when I can review again